packages/nimble-components/src/rich-text/components/mention-view/mention-users-view/template.ts

```diff
@@ -0,0 +1,11 @@
+import { html } from '@microsoft/fast-element';
+import type { RichTextMentionUsersView } from '.';
+
+export const template = html<RichTextMentionUsersView>`<span
+    class="control"
+    part="control"
+    mention-href="${x => x.mentionHref}"
+    mention-label="${x => x.mentionLabel}"
+>
+    <slot></slot>
```

Comment:
What is the component trying to abstract? I though the attributes would be the only configuration, i.e. the mention-href and the mention-label. And those attributes would be used to populate the internal contents, i.e. show the `@` character and the user name. Why is the slot needed?

Reply:
Used the slot approach for a couple of reasons:
1. Tiptap typically renders the mention node in the default slot of the span element through the [`renderLabel`](https://tiptap.dev/api/nodes/mention#render-label) configuration by accessing the node values. So I tried to make our custom component behave the same as the default `span` element does.
2. A workaround for the Firefox bug: The bug is related to the selection of text within the shadow root element. Links to the bugs,
    1. https://bugzilla.mozilla.org/show_bug.cgi?id=1455893 
    2. https://bugzilla.mozilla.org/show_bug.cgi?id=1545367
    3. https://bugzilla.mozilla.org/show_bug.cgi?id=1860292

   All the above bugs are related which states that there is an issue in selecting the texts when the content is within a shadow root. In our case, it affects in below scenarios when opened in the Firefox browser,
    1. In the editor, when we select the mention node, the blue selection background is not shown
    2. In the viewer, not able to select both the @mention view node and other texts in the viewer at the same time

The above issues are resolved when using a slot to render the name from the parent component.

However, this slot approach creates other new issues in the selection of texts in the editor and they are:
1. Selecting the mention view node using the keyboard (Shift + left/right arrow keys) doesn't work as expected
2. If there is no whitespace/any other character at the beginning and at the end of the mention view node in the editor, keyboard selection doesn't work but mouse selection works

1 is resolved when populating the `mention-label` as text content with `@`. But 2 is not resolved in both cases. So both approach has some issues, thought that the Firefox issue was specific to a single browser and the other one was generic, so I have updated the code by populating the attribute value. We are finding it a bit hard to render a shadow root component within a `contenteditable` div across different browsers that creates a different behavioral change like the issues we faced when rendering [`nimble-anchor` within the editor](https://github.com/ni/nimble/issues/1516).

Reply:
I'm not clear on a couple things. My primary question is trying to understand what view is rendered while the mention is being entered vs after the entering is complete.

For example while entering I expect the characters the user is entering to be emitted into the slot but after an option is selected maybe there would be a completely custom view.

For example like in GitHub I can type the characters `#94` and select the corresponding option but it gets rendered with a custom view: 
![image](https://github.com/ni/nimble/assets/1588923/89211779-f5de-4fa0-905e-7919eaa6187a)

The current component being proposed doesn't make it clear to me how the view after entered would be customized.
A thought is maybe something like:

During editing:
```
<rich-text-mention-users-view is-editing></rich-text-mention-users-view>
```
Behavior:
Text is inserted into the slot, all the element does is style the text in the slot.

After mention is commited:
```
<rich-text-mention-users-view mention-href="url" mention-label="John Doe"></rich-text-mention-users-view>
```
Behavior:
Text in the slot is configured as display:none, The configured attributes are used to render the content in the shadow dom template using the `@` icon and formatted text. In the future may be hover buttons, card, etc that display a lot of content or maybe it becomes a link, etc.

Reply:
> the mention is being entered vs after the entering is complete.

This is possible by modifying the [`decorationTag`](https://tiptap.dev/api/utilities/suggestion#decoration-tag) during editing with the `editing` boolean attribute and [`renderHtml`](https://tiptap.dev/guide/custom-extensions/#render-html) for rendering the view node after mention is committed without the `editing` boolean attribute in the Titpap editor.

I have updated the code as suggested by having a `@` icon in the view mode. I just updated the color of the icon by updating the `--ni-nimble-icon-color` to match the text color. Please let me know if that is fine.

Reply:
Or can it be inverted? Looks like render html does let you set attributes, so we could make it `disable-editing`

Reply:
Yes, `renderHtml` allows us to set attributes. I will update the code shortly.

Reply:
I have updated the code with `disable-editing` attribute and verified its working in the editor. Kindly review the same.